src/AuthenticationService.php

```diff
@@ -190,7 +194,12 @@ public function authenticate(ServerRequestInterface $request): ResultInterface
         $result = null;
         /** @var \Authentication\Authenticator\AuthenticatorInterface $authenticator */
         foreach ($this->authenticators() as $authenticator) {
-            $result = $authenticator->authenticate($request);
+            $result = $this->dispatchAuthenticateEvent(
```

Comment:
Is it ok if this trigger on every request? Perhaps we should include that in the documentation?